### DISCUSSION TITLE
Check that the scene passed to AddConnectionToScene is valid

### DISCUSSION THREAD

**Discussion by mifozski** [FirstGearGames/FishNet#805] [Feature Request]
The idea is to check if the scene passed down to AddConnectionToScene is valid. A scene instance can be invalid, i.e. referencing a scene that does not exist. AddConnectionToScene eats it up and doesn't log any errors or messages. This leads to very weird bugs that happen relatively far down the path on the client side, and that are hard to track back to the AddConnectionToScene call.

Background:
I spent about two days trying to understand why the object syncing for a dynamically loaded scene was not working.
It turned out I was calling AddConnectionToScene incorrectly. I was using SceneManager from UnityEngine to retrieve the scene but instead of passing the name I was passing the path, and since SceneManager just silently returned an invalid scene, it took some time to figure this out.

**Comment by FirstGearGames** [comment]
I added this to the method in 4.5.4
```cs
            if (!scene.isLoaded || !scene.IsValid())
            {
                NetworkManager.LogError($"Only valid, loaded scenes may be used.");
                return;
            }
```
